src/lib/integration_test_lib/graphql_requests.ml

```diff
@@ -0,0 +1,1013 @@
+  let sender_pk_str =
+    Signature_lib.Public_key.Compressed.to_string sender_pub_key
+  in
+  [%log info] "send_delegation: unlocking account"
+    ~metadata:[ ("sender_pk", `String sender_pk_str) ] ;
+  let unlock_sender_account_graphql () =
+    let unlock_account_obj =
+      Graphql.Unlock_account.(
+        make
+        @@ makeVariables ~password:"naughty blue worm"
+             ~public_key:sender_pub_key ())
+    in
+    exec_graphql_request ~logger ~node_uri
+      ~query_name:"unlock_sender_account_graphql" unlock_account_obj
+  in
+  let%bind _ = unlock_sender_account_graphql () in
+  let send_delegation_graphql () =
+    let input =
+      Mina_graphql.Types.Input.SendDelegationInput.make_input
+        ~from:sender_pub_key ~to_:receiver_pub_key ~fee ()
+    in
+    let send_delegation_obj =
+      Graphql.Send_delegation.(make @@ makeVariables ~input ())
+    in
+    exec_graphql_request ~logger ~node_uri ~query_name:"send_delegation_graphql"
+      send_delegation_obj
+  in
+  let%map result_obj = send_delegation_graphql () in
+  let return_obj = result_obj.sendDelegation.delegation in
+  let res =
+    { id = transaction_id_to_string return_obj.id
+    ; hash = return_obj.hash
+    ; nonce = Mina_numbers.Account_nonce.of_int return_obj.nonce
+    }
+  in
+  [%log info] "stake delegation sent"
+    ~metadata:
+      [ ("user_command_id", `String res.id)
+      ; ("hash", `String (Transaction_hash.to_base58_check res.hash))
+      ; ("nonce", `Int (Mina_numbers.Account_nonce.to_int res.nonce))
+      ] ;
+  res
+
+let must_send_delegation ~logger node_uri ~sender_pub_key
+    ~(receiver_pub_key : Account.key) ~fee =
+  send_delegation ~logger node_uri ~sender_pub_key ~receiver_pub_key ~fee
+  |> Deferred.bind ~f:Malleable_error.or_hard_error
+
+let send_payment_with_raw_sig ~logger node_uri ~sender_pub_key ~receiver_pub_key
+    ~amount ~fee ~nonce ~memo
+    ~(valid_until : Mina_numbers.Global_slot_since_genesis.t) ~raw_signature =
+  [%log info] "Sending a payment with raw signature"
+    ~metadata:
+      [ ( "sender_pub_key"
+        , Signature_lib.Public_key.Compressed.to_yojson sender_pub_key )
+      ; ("node_uri", `String (Uri.to_string node_uri))
+      ] ;
+  let open Deferred.Or_error.Let_syntax in
+  let send_payment_graphql () =
+    let open Graphql.Send_payment_with_raw_sig in
+    let input =
+      Mina_graphql.Types.Input.SendPaymentInput.make_input ~from:sender_pub_key
+        ~to_:receiver_pub_key ~amount ~fee ~memo ~nonce
+        ~valid_until:
+          (Mina_numbers.Global_slot_since_genesis.to_uint32 valid_until)
+        ()
+    in
+    let variables = makeVariables ~input ~rawSignature:raw_signature () in
+    let send_payment_obj = make variables in
+    let variables_json_basic = variablesToJson (serializeVariables variables) in
+    (* An awkward conversion from Yojson.Basic to Yojson.Safe *)
+    let variables_json =
+      Yojson.Basic.to_string variables_json_basic |> Yojson.Safe.from_string
```

Comment:
Yuck. `(variables_json_basic :> Yojson.Safe.t)` would be better.

Reply:
this whole codeblock is just for printing out diagnostics.  i changed it to just printing out each of the variables one by one to avoid ugly conversions